src/theory/builtin/theory_builtin_rewriter.h

```diff
@@ -56,6 +80,17 @@ class TheoryBuiltinRewriter {
   static inline void init() {}
   static inline void shutdown() {}
 
+// conversions between lambdas and arrays
+private:  
+  static Node getLambdaForArrayRepresentationRec( TNode a, TNode bvl, unsigned bvlIndex, 
+                                                  std::unordered_map< TNode, Node, TNodeHashFunction >& visited );
+public:
+  /** given an array constant a, returns a lambda expression that it corresponds to, with bound variable list bvl. */
+  static Node getLambdaForArrayRepresentation( TNode a, TNode bvl );
+  /** given a lambda expression n, returns an array term. reqConst is true if we require the return value to be a constant. */
+  static Node getArrayRepresentationForLambda( TNode n, bool reqConst = false );
+  /** get a canonical bound variable list for function type tn */
+  static Node getLambdaBoundVarListForType( TypeNode tn, unsigned nargs );
```

Comment:
This seems like it belongs somewhere else, in the node manager perhaps?

Reply:
At the moment, I prefer having this in the builtin rewriter, since it is indicative that this conversion is only used internally to check whether a lambda is constant. We will by design not be doing conversions to and from arrays at the solving level.

Reply:
OK, I agree for the conversion methods, but what about the getLambdaBoundVarListForType.  Seems like this could be changed to just getBoundVarListForFunctionType and become a general-purpose NodeManager method?

Reply:
I've moved this to NodeManager in my latest commit.